test/functional/test_framework/util.py

```diff
@@ -226,7 +226,7 @@ def ceildiv(a, b):
 def get_fee(tx_size, feerate_btc_kvb):
     """Calculate the fee in BTC given a feerate is BTC/kvB. Reflects CFeeRate::GetFee"""
     feerate_sat_kvb = int(feerate_btc_kvb * Decimal(1e8)) # Fee in sat/kvb as an int to avoid float precision errors
-    target_fee_sat = ceildiv(feerate_sat_kvb * tx_size, 1000) # Round calculated fee up to nearest sat
+    target_fee_sat = ceildiv(feerate_sat_kvb * int(tx_size), 1000) # Round calculated fee up to nearest sat
```

Comment:
In commit "test: Call ceildiv helper with integer" (11d4385819154e637eb34879bf3e4040f85ef194)

Nice find!

This seems like a working fix but fragile. It seems bad if callers are passing transaction sizes as decimals that will get silently rounded, instead of just passing integer values. I think it would be better to drop the rounding cast, and write `assert isinstance(tx_size, int)`, then fix the bad callers which are passing fractional transaction sizes.

I think it would also be good to add `assert isinstance(a, int)` and `assert isinstance(b, int)` to the ceildiv function with a comment like "# Implementation requires python integers, which have a // operator that does floor division. Other types like decimal.Decimal whose // operator truncates towards 0 will not work."

Reply:
Thanks, that makes sense. I added the asserts as suggested and also your explanation.
I changed the call sites to use the `count_bytes` helper for the tx size, so that the function is called with integers now.